File: Projects/C/Beginner-Projects/todo-list/todo-1/todo.c
#include <stdio.h>
#include <string.h>

#define MAX_TASKS 100
#define MAX_LENGTH 100

struct Task {
    char description[MAX_LENGTH];
    int completed;
};

struct Task todoList[MAX_TASKS];
int taskCount = 0;

void addTask() {
    if(taskCount >= MAX_TASKS) {
        printf("Task list is full!\n");
        return;
    }
    
    printf("Enter task description: ");
    getchar(); // Clear newline character
    fgets(todoList[taskCount].description, MAX_LENGTH, stdin);
    
    // Remove newline character
    todoList[taskCount].description[strcspn(todoList[taskCount].description, "\n")] = '\0';
    
    todoList[taskCount].completed = 0;
    taskCount++;
    
    printf("Task added successfully!\n");
}

void displayTasks() {
    if(taskCount == 0) {
        printf("No tasks to display.\n");
        return;
    }
    
    printf("\n=== YOUR TASKS ===\n");
    for(int i = 0; i < taskCount; i++) {
        printf("%d. [%s] %s\n", 
               i + 1, 
               todoList[i].completed ? "X" : " ", 
               todoList[i].description);
    }
    printf("==================\n\n");
}

void markComplete() {
    int taskNum;
    
    if(taskCount == 0) {
        printf("No tasks to complete.\n");
        return;
    }
    
    printf("Enter task number to mark complete: ");
    scanf("%d", &taskNum);
    
    if(taskNum >= 1 && taskNum <= taskCount) {
        todoList[taskNum - 1].completed = 1;
        printf("Task marked as complete!\n");
    } else {
        printf("Invalid task number!\n");
    }
}

void displayMenu() {
    printf("=== TODO LIST MANAGER ===\n");
    printf("1. Add Task\n");
    printf("2. View Tasks\n");
    printf("3. Mark Task Complete\n");
    printf("4. Exit\n");
    printf("Choose an option: ");
}

int main() {
    int choice;
    
    printf("Welcome to TODO List Manager!\n\n");
    
    do {
        displayMenu();
        scanf("%d", &choice);
        
        switch(choice) {
            case 1:
                addTask();
                break;
            case 2:
                displayTasks();
                break;
            case 3:
                markComplete();
                break;
            case 4:
                printf("Goodbye!\n");
                break;
            default:
                printf("Invalid choice! Please try again.\n");
        }
        
        printf("\n");
    } while(choice != 4);
    
    return 0;
}

File: Projects/Python/Beginner-Projects/todo-list/todo-1/todo.py
# 📝 To-Do List - Level 1 (Memory Track)
# Welcome to a new project! 
# A To-Do list is the classic "Hello World" of intermediate programming. 
# In Level 1, we use pure Python. ZERO imported libraries!

def main():
    # This is a Python List. It will hold our tasks in the computer's RAM (memory).
    # NOTE: Because it's just in memory, if you exit the program, the tasks vanish!
    tasks = []

    print("========================================")
    print("      📝 MY SIMPLE TO-DO LIST 📝      ")
    print("========================================")

    # We use a 'while True' loop to keep asking the user what they want to do.
    while True:
        print("\n--- Options ---")
        print("1. View Tasks")
        print("2. Add a Task")
        print("3. Complete/Remove a Task")
        print("4. Exit")
        
        choice = input("\nWhat would you like to do? (1-4): ").strip()

        if choice == '1':
            print("\n--- Your Tasks ---")
            # If the list is empty, tell them.
            if len(tasks) == 0:
                print("Your list is empty! Go relax! ☕")
            else:
                # Loop through the list and print each task.
                # 'enumerate' is a cool trick that gives us both the index (number) and the task itself!
                # We do start=1 so the list looks like [1, 2, 3] instead of Python's default [0, 1, 2]
                for index, task in enumerate(tasks, start=1):
                    print(f"[{index}] {task}")
                    
        elif choice == '2':
            # Add a new task to the end of our list using .append()
            new_task = input("Enter the new task: ").strip()
            if new_task:
                tasks.append(new_task)
                print(f"✅ Added: '{new_task}'")
            else:
                print("⚠️ You cannot add an empty task!")

        elif choice == '3':
            # To delete a task, we need to know its number.
            if len(tasks) == 0:
                print("There's nothing to cross off yet!")
                continue

            try:
                task_num = int(input("Enter the number of the task you completed: "))
                # Humans count from 1, but Python lists start at index 0.
                # So we subtract 1 to find the correct item in the list!
                index_to_remove = task_num - 1
                
                # Check if the number is valid
                if 0 <= index_to_remove < len(tasks):
                    removed_task = tasks.pop(index_to_remove)
                    print(f"🎉 Well done! You finished: '{removed_task}'")
                else:
                    print("⚠️ That number is not on the list!")
            except ValueError:
                print("⚠️ Please enter a valid number, not text.")

        elif choice == '4':
            # Break out of the infinite loop
            print("Goodbye! See you next time! 👋")
            break
            
        else:
            print("⚠️ Invalid choice. Please press 1, 2, 3, or 4.")

# Standard entry point execution
if __name__ == "__main__":
    main()
